src/containers/Tenant/Query/QuerySettingsDialog/QuerySettingsTimeout.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import {CircleQuestion} from '@gravity-ui/icons';
+import {Icon, Popover, Switch, TextInput} from '@gravity-ui/uikit';
+
+import {cn} from '../../../../utils/cn';
+import {ENABLE_QUERY_STREAMING} from '../../../../utils/constants';
+import {useSetting} from '../../../../utils/hooks';
+
+import {QUERY_SETTINGS_FIELD_SETTINGS} from './constants';
+import i18n from './i18n';
+
+import './QuerySettingsTimeout.scss';
+
+const b = cn('ydb-query-settings-timeout');
+
+interface QuerySettingsTimeoutProps {
+    id?: string;
+    value: number | null | undefined;
+    onChange: (value: number | undefined) => void;
+    onToggle: (enabled: boolean) => void;
+    validationState?: 'invalid';
+    errorMessage?: string;
+    isDisabled?: boolean;
+}
+
+export function QuerySettingsTimeout({
+    id,
+    value,
+    onChange,
+    onToggle,
+    validationState,
+    errorMessage,
+    isDisabled,
+}: QuerySettingsTimeoutProps) {
+    const [isQueryStreamingEnabled] = useSetting(ENABLE_QUERY_STREAMING);
+
+    const handleValueChange = React.useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>) => {
+            const newValue = event.target.value ? Number(event.target.value) : undefined;
+            onChange(newValue);
+        },
+        [onChange],
+    );
+
+    const isChecked = value !== null;
+
+    const queryStreamingLabel = isQueryStreamingEnabled ? (
```

Comment:
maybe rename to `timeoutLabel`? And I would do it a separate component (not a demand, just suggestion).